src/org/scada_lts/dao/event/UserEventDAO.java

```diff
@@ -0,0 +1,171 @@
+	private static final String COLUMN_NAME_EVENT_ID = "eventId";
+	private static final int    COLUMN_INDEX_EVENT_ID =  1;
+	private static final String COLUMN_NAME_USER_ID = "userId";
+	private static final int    COLUMN_INDEX_USER_ID = 2;
+	
+	// @formatter:off
+	
+	private static final String USER_EVENT_SELECT=""	
+			+"select "
+				+COLUMN_NAME_EVENT_ID+","
+				+COLUMN_NAME_SILENCED+","
+				+COLUMN_NAME_USER_ID+" "
+			+"from "+
+				"userevents ";
+	
+	private static final String USER_EVENT_SELECT_BASE_ON_PK = ""+
+			USER_EVENT_SELECT
+			+"where "
+				+COLUMN_NAME_EVENT_ID+"=? and "
+				+COLUMN_NAME_USER_ID+"=?";
+	
+	private static final String USER_EVENT_INSERT="" +
+			"insert userEvents ("
+				+COLUMN_NAME_EVENT_ID+","
+				+COLUMN_NAME_USER_ID+","
+				+COLUMN_NAME_SILENCED+") "
+			+ "values (?,?,?)";
+	
+	private static final String USER_EVENT_ACK ="" +
+			"update "
+				+"userEvents set "
+				+ COLUMN_NAME_SILENCED+"=? " 
+			+"where "
+				+ COLUMN_NAME_EVENT_ID+"=?";
+	
+	// @formatter:onn
+	
+	// RowMapper
+	private class UserEventRowMapper implements RowMapper<UserEvent> {
+		public UserEvent mapRow(ResultSet rs, int rowNum) throws SQLException {
+			UserEvent userEvent;
+
+			userEvent = new UserEvent();
+			userEvent.setEventId(rs.getLong(COLUMN_NAME_EVENT_ID));
+			userEvent.setUserId(rs.getLong(COLUMN_NAME_USER_ID));
+			userEvent.setSilenced(DAO.charToBool(rs.getString(COLUMN_NAME_SILENCED)));
+						
+			return userEvent;
+			
+		}
+	}
+
+
+	@Override
+	public List<UserEvent> findAll() {
+		return (List<UserEvent>) DAO.getInstance().getJdbcTemp().query(USER_EVENT_SELECT, new Object[]{ }, new UserEventRowMapper());
+	}
+
+	@Override
+	public UserEvent findById(Object[] pk) {
+		return ((List<UserEvent>) DAO.getInstance().getJdbcTemp().query(USER_EVENT_SELECT_BASE_ON_PK, pk, new UserEventRowMapper())).get(0);
+	}
+
+	@Override
+	public List<UserEvent> filtered(String filter, Object[] argsFilter, long limit) {
+		String myLimit="";
+		Object[] args;
+		if (limit != NO_LIMIT) {
+			myLimit = LIMIT+" ? ";
+			args = DAO.getInstance().appendValue(argsFilter, String.valueOf(limit));
+		} else {
+			args=argsFilter;
+		}
+	
+		return (List<UserEvent>) DAO.getInstance().getJdbcTemp().query(USER_EVENT_SELECT+" where "+ filter + myLimit, args, new UserEventRowMapper());
+	}
+
+	@Override
+	public Object[] create(UserEvent entity) {
+		
+		if (LOG.isTraceEnabled()) {
+			LOG.trace(entity);
+		}
+		
+		
+		
+		DAO.getInstance().getJdbcTemp().update(USER_EVENT_INSERT,  new Object[] { 
+			 						entity.getEventId(),
+			 						entity.getUserId(),
+			 						DAO.boolToChar(entity.isSilenced())
+			 				});
+			 				
+		
+		return new Object[] {entity.getEventId(), entity.getUserId()};
+		
+	}
+	
+	public void batchUpdate(final int eventId,	final List<Integer> userIds, final boolean alarm) {
+		DAO.getInstance().getJdbcTemp().batchUpdate(USER_EVENT_INSERT, new BatchPreparedStatementSetter() {
```

Comment:
This line throws "java.sql.BatchUpdateException" with a message "Cannot add or update a child row: a foreign key co
nstraint fails (`scadalts`.`userevents`, CONSTRAINT `userEventsFk1` FOREIGN KEY (`eventId`) REFERENCES `events` (`id`) ON DELETE CASCADE);". You should investigate this error and fix it.